src/ThemedControls.cpp

```
@@ -249,16 +538,82 @@ void ThemedChoice::OnPaint(wxPaintEvent&) {
+}
+
+void ThemedChoice::OpenPopup() {
   if (!IsEnabled() || m_items.empty()) return;
-  wxMenu menu;
-  for (int i = 0; i < static_cast<int>(m_items.size()); ++i)
-    menu.AppendCheckItem(1000 + i, m_items[i].label);
-  if (m_selection >= 0) menu.Check(1000 + m_selection, true);
-  const int sel = GetPopupMenuSelectionFromUser(menu);
-  if (sel == wxID_NONE) return;
-  SetSelection(sel - 1000);
+  if (m_open_popup) {  // clicked/pressed again while open: toggle it closed
```

Clang (14.0.6)

[warning] implicit conversion 'ThemedChoicePopup *' -> bool

(readability-implicit-bool-conversion)